Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs

```diff
@@ -112,8 +122,29 @@ public void ExitDisposals(EntityUid uid, DisposalHolderComponent? holder = null,
                 if (duc != null)
                     duc.Container.Insert(entity, EntityManager, xform, meta: meta);
                 else
-                    _xformSystem.AttachToGridOrMap(entity, xform);
+                {
+                    // fragile things go splat
+                    if (TryComp<FragileComponent>(entity, out var fragile))
```

Comment:
If you're going to implement this through a new component, please don't add an explicit check for that component. Use an event instead, that way if thirty people decide they want to make entities do something when the come out of a disposal chute this section doesn't with up 300 lines long.

Reply:
@deltanedas explicitly said to use TryComp which is why I used it. Can you point me at an example where something similar is being done with events?

Reply:
Sure.
For lightbulbs breaking on impact when thrown we have as follows:
1. When thrown objects land the function [raises an event](https://github.com/space-wizards/space-station-14/blob/367c251e09c1c4be2c89e69fa0f08cfbcb756d77/Content.Shared/Throwing/ThrownItemSystem.cs#L130-#L131) ([defined here](https://github.com/space-wizards/space-station-14/blob/master/Content.Shared/Throwing/LandEvent.cs)).
2. Somewhere in the [system that handles light bulbs](https://github.com/space-wizards/space-station-14/blob/367c251e09c1c4be2c89e69fa0f08cfbcb756d77/Content.Server/Light/EntitySystems/LightBulbSystem.cs) we have a [function that handles beaking lightbulbs on impact](https://github.com/space-wizards/space-station-14/blob/367c251e09c1c4be2c89e69fa0f08cfbcb756d77/Content.Server/Light/EntitySystems/LightBulbSystem.cs#L31-#L35).
3. When the light bulb system starts up we [tell the engine we want the lightbulb-breaking handler to be run when something happens](https://github.com/space-wizards/space-station-14/blob/367c251e09c1c4be2c89e69fa0f08cfbcb756d77/Content.Server/Light/EntitySystems/LightBulbSystem.cs#L20)

Reply:
i said trycomp as it would be better than manual prototype component lookup also entities can exist without prototypes

Reply:
So you're saying within `ExitDisposals` I would raise an OnExitDisposals event. Within the system that handles disposals, I would have a function that slightly throws the thing. When the disposals system starts up I would tell the engine that I want the on exit handler to be run with something happens.

Reply:
If you decide to just throw the items when the exit the disposal chute _that_ is probably fine to hardcode, since it would be a behaviour of the disposal chute itself instead of the item. The event route is If you want your component to react specifically to exiting disposals then you would use the event route.

Reply:
Got it. I'll remove my component and go with the "everything is getting thrown upon exiting" hardcode.